extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/Common.java

```diff
@@ -31,7 +31,11 @@
 
   private Common() {}
 
-  static SpanContext buildSpanContext(String traceId, String spanId, String sampled) {
+  static SpanContext buildSpanContext(
+      @Nullable String traceId, @Nullable String spanId, @Nullable String sampled) {
+    if (traceId == null || spanId == null) {
```

Comment:
I wouldn't add these if I added `Contract` to `isTraceIdValid` and `isSpanIdValid`. As far as I know there isn't any performance penalty for this sort of null check and it seems OK to just do this instead.